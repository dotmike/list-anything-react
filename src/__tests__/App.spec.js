import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import ConnectedApp, { UnconnectedApp } from "../App";

const defaultProps = {
  history: {
    push: () => {}
  },
  location: {
    pathname: ""
  }
};

describe("ConnectedApp", () => {
  const mockStore = configureStore([thunk]);
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedApp {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedApp", () => {
  const props = {
    ...defaultProps,
    isAuthenticated: false,
    fetchAllSubjects: () => {}
  };

  describe("when `isAuthenticated` is false", () => {
    it("renders correctly", () => {
      props.isAuthenticated = false;

      const wrapper = shallowWithContext(<UnconnectedApp {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
