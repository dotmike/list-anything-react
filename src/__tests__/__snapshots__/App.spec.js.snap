// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedApp renders correctly 1`] = `
<App
  fetchAllSubjects={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  isAuthenticated={false}
  location={
    Object {
      "pathname": "",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedApp when \`isAuthenticated\` is false renders correctly 1`] = `
<div
  id="app-container"
>
  <Route
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "",
      }
    }
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "",
      }
    }
    path="/confirmation/:token"
  />
  <PrivateRoutes
    history={
      Object {
        "push": [Function],
      }
    }
    location={
      Object {
        "pathname": "",
      }
    }
    showContent={false}
  />
  <PublicRoutes
    location={
      Object {
        "pathname": "",
      }
    }
    showContent={true}
  />
</div>
`;
